### PR TITLE
Update error filtering and deprecation methods

### PR DESCRIPTION
**Pull request by cortze**
# Motivation
There are still flaky results on the deprecation system and info organization on the db organization.
 
# Description
This PR aims to clean those uncertainties improving our possibility to detect and deprecate peers

# Tasks
- [x] Filter the possible connection errors
- [x] Update Chain details only when they are not null

# Proof of Success

### CODE REVIEW

File: crawler/peering.go [hunk 1]
```diff
@@ -114,7 +112,10 @@ func (p *Peering) runOrcherster() {
...
 					continue
 				}
-				p.dialC <- nextNode.hostInfo
+				isTimeToDial := nextNode.nextDialTime.Before(time.Now())
+				if isTimeToDial {
+					p.dialC <- nextNode.hostInfo
+				}
 				dialedCache[nextNode.hostInfo.ID] = struct{}{}
 			} else {
 				// still check the contexts in case we have to interrupt
```
Comment (tdahar) [inline crawler/peering.go:121]: What is supposed to happen in the case `<-startT.C`

File: models/node.go [hunk 2]
```diff
@@ -118,3 +118,7 @@ type ChainDetails struct {
 	NetworkID       uint64
 	TotalDifficulty *big.Int
 }
+
+func (d *ChainDetails) IsEmpty() bool {
+	return d.NetworkID == uint64(0)
```
Comment (tdahar) [inline models/node.go:123]: I know it is very simple, but it might be a good idea to have a constant for an empty chain

**Review by tdahar — APPROVED** [review]
In general it looks very good :) I am just missing some more description of what you had to do in the Pull Request description and some proof (it can be a screenshot of the db or anything)
Anyways, very nice job!

File: crawler/peering.go (hunk 1 above)
Comment (santi1234567) [reply crawler/peering.go:121]: I believe that is a delay for when there are no nodes initially, to not loop constantly.

**Comment by santi1234567** [comment]
# Update on PR
@cortze added error parsing for connections and the handling of those connections in terms of dialing timers.

I completed the implementation of dial timers since they weren't being used (the script dialed nonstop).

I also fixed a few existing a new bugs product of changes in this PR:
- Chain details not updating because of conflict on upsert. (In reality, only an update was needed).
- Filtering for nodes that werent' in the same network on failed connections meant that nodes that there was no room for redialing to failed connections that were on the same network.
-  `DialedCache` (map of already dialed nodes on the same loop) wasn't updating so the system could never redial on the same run of the script.
- `IsDeprecable` had flawed logic since the deprecation timer wasn't being used.

**Comment by santi1234567** [comment]
# Proof of success
image

File: crawler/peering.go (hunk 1 above)
Comment (tdahar) [reply crawler/peering.go:121]: Cool :) That makes sense, thanks for the explanation

**Comment by tdahar** [comment]
Cool! Thanks for these comments, I think it makes it more clear what you have fixed / done in this PR :)
Nice job!

**Review by tdahar — APPROVED** [review]
LGTM!

**Pull request merged**